pages/contributors.js

```diff
@@ -0,0 +1,220 @@
+import axios from "axios"
+import React from "react"
+import Head from "next/head"
+import { useEffect, useState } from "react"
+import animationData from "../lottie/sad.json"
+import { Text } from "atomize"
+import { Spinner } from "react-bootstrap"
+import Lottie from "react-lottie"
+import Card from "../components/Cards/Card"
+
+const defaultOptions = {
+    loop: true,
+    autoplay: true,
+    animationData: animationData,
+}
+
+export default function Contributors() {
+    const [contriFrontend, setContributorsFront] = useState()
+    const [contriBackend, setContributorsBack] = useState()
+    const [contributors, setContributors] = useState()
+    const [error, setError] = useState([false, false, false])
+
+    useEffect(() => {
+        axios
+            .get(
+                "https://api.github.com/repos/COPS-IITBHU/hackalog-frontend/contributors"
+            )
+            .then((res) => setContributorsFront(res.data))
+            .catch((err) => {
+                let arr = error
+                arr[0] = true
+                setError(arr)
+            })
+        axios
+            .get(
+                "https://api.github.com/repos/COPS-IITBHU/hackalog-backend/contributors"
+            )
+            .then((res) => setContributorsBack(res.data))
+            .catch((err) => {
+                let arr = error
+                arr[1] = true
+                setError(arr)
+            })
+    }, [error])
+
+    if (!contributors) {
+        if (contriFrontend && contriBackend) {
+            let commonF = []
+            let commonB = []
+            let arr = []
+            for (let x of contriFrontend) {
+                for (let y of contriBackend) {
+                    if (x.login === y.login) commonF.push(x), commonB.push(y)
```

Comment:
since you are adding the same elements to both the arrays, I don't think there is a need to create 2 arrays namely `commonF` and `commonB`, just creating 1 array `common` can also do the job

Reply:
actually the Objects in the Backend and Frontend for the same user are different in the contributions key, so 2 different arrays need to be created, Also the push into the arrays are different

Reply:
but still the meta data that we are using, such as `login` `avatar_url` `html_url`..., are still the same, hence it's just redundant data.

Reply:
to replace `commonF` and `commonB` with single `common` array, I will have to loop through the `contriFrontend` and `contriBackend` arrays and change them to keep only the necessary keys and kinda remove the different ones, but this process will also require making of a new Array of those objects.
There would be another way with just one `common` array but it would require to run loops within loop to ensure that common cards are not created as the arrays have objects and cant be accessed directly.

So, Please tell how I should proceed.

Reply:
- Since you are passing the strings "hackalog-frontend" and "hackalog-backend" in the description to check which repositories the user has contributed to, we can instead make two boolean fields `backendContributor` and `frontendContributor`. (Or maybe an array of two booleans)
- Now, we may set the backend field to `true` if the user has contributed to the backend. Same for the front end.
- Finally, we render this conditionally in the card like this:
```js
<h4>Contributed to:</h4>
{
    backend && <a
    href="https://github.com/COPS-IITBHU/hackalog-frontend"
    className="text-muted"
>hackalog-backend</a>
}
```
- Now, you won't require two separate arrays for the purpose.

This is one approach. Another one would be to store the user information in a hashmap and access it using their unique key (username in this case).